From: Chuck Short <zulcss@ubuntu.com>
Date: Tue, 24 Jul 2007 23:30:06 +0000 (-0400)
Subject: UBUNTU: SAUCE: [USB]: add ASUS LCM to the blacklist
X-Git-Url: http://kernel.ubuntu.com/git?p=ubuntu%2Fubuntu-intrepid.git;a=commitdiff_plain;h=e86ba19eb3d99a888825031c9c43f729c668e65d

UBUNTU: SAUCE: [USB]: add ASUS LCM to the blacklist

from linux-usb-devel

Some of ASUS' notebooks (e.g G Series) include a tiny oled display, which is
attached to an internal USB bus. Unfortunatly the device reports a wrong
DeviceDescriptor and is therefore identified as a HID device.

Signed-off-by: Chuck Short <zulcss@ubuntu.com>
---

Index: 2008-devel/drivers/hid/usbhid/hid-quirks.c
===================================================================
--- 2008-devel.orig/drivers/hid/usbhid/hid-quirks.c
+++ 2008-devel/drivers/hid/usbhid/hid-quirks.c
@@ -86,6 +86,9 @@
 #define USB_VENDOR_ID_ASUS		0x0b05
 #define USB_DEVICE_ID_ASUS_LCM		0x1726
 
+#define USB_VENDOR_ID_ASUS		0x0b05
+#define USB_DEVICE_ID_ASUS_LCM		0x1726
+
 #define USB_VENDOR_ID_ATEN		0x0557
 #define USB_DEVICE_ID_ATEN_UC100KM	0x2004
 #define USB_DEVICE_ID_ATEN_CS124U	0x2202
@@ -744,6 +747,8 @@ static const struct hid_rdesc_blacklist 
 	{ USB_VENDOR_ID_CYPRESS, USB_DEVICE_ID_CYPRESS_BARCODE_1, HID_QUIRK_RDESC_SWAPPED_MIN_MAX },
 	{ USB_VENDOR_ID_CYPRESS, USB_DEVICE_ID_CYPRESS_BARCODE_2, HID_QUIRK_RDESC_SWAPPED_MIN_MAX },
 
+	{ USB_VENDOR_ID_ASUS, USB_DEVICE_ID_ASUS_LCM, HID_QUIRK_IGNORE},
+
 	{ 0, 0 }
 };
 
